File: ProjectM/app/src/main/java/com/example/majorproject/ApplicationClass.java
package com.example.majorproject;

import android.app.Application;
import com.onesignal.OneSignal;
import com.onesignal.debug.LogLevel;
import com.onesignal.Continue;

public class ApplicationClass extends Application {
    private static final String ONESIGNAL_APP_ID = "68474ac2-e2f4-4ed3-808e-c0647ea7bd56";

    @Override
    public void onCreate() {
        super.onCreate();

        // Verbose Logging set to help debug issues, remove before releasing your app.
        OneSignal.getDebug().setLogLevel(LogLevel.VERBOSE);

        // OneSignal Initialization
        OneSignal.initWithContext(this, ONESIGNAL_APP_ID);

        // requestPermission will show the native Android notification permission prompt.
        // NOTE: It's recommended to use a OneSignal In-App Message to prompt instead.
        OneSignal.getNotifications().requestPermission(false, Continue.none());

    }
}

File: ProjectM/app/src/main/java/com/example/majorproject/ApplicationClass.kt
package com.example.majorproject

import android.app.Application
import android.util.Log
import com.example.majorproject.notification.NotificationHelper

class ApplicationClass : Application() {

    override fun onCreate() {
        super.onCreate()

        val notificationHelper = NotificationHelper()
        notificationHelper.createChannel(this) // Ensure the channel is created before sending notifications

        // Try sending a notification
        try {
            notificationHelper.sendNotification(
                applicationContext,
                "New Message",
                "You have a new message waiting!",
                R.drawable.banner4
            )
        } catch (e: Exception) {
            Log.e("ApplicationClass", "Error sending notification", e)
        }
    }
}
